styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -285,7 +288,24 @@ private static EnvVar envVar(String name, String value) {
 
   @Override
   public void cleanup(String executionId) {
-    client.pods().withName(executionId).delete();
+    final Pod pod = client.pods().withName(executionId).get();
+    if (pod != null) {
+      cleanup(pod);
```

Comment:
Yeah, that's when state machine transits which means the pod is a valid one and executionId is the pod name. This however handles some corner cases for example pod gone for no reason or being manually deleted after generating event and before transiting state. It's OK.